sklearn/model_selection/tests/test_validation.py

```diff
@@ -713,6 +738,25 @@ def test_learning_curve_with_boolean_indices():
                               np.linspace(0.1, 1.0, 10))
 
 
+def test_learning_curve_with_shuffle():
+    X = np.array([[1, 2], [3, 4], [5, 6], [7, 8], [11, 12], [13, 14], [15, 16],
+                 [17, 18], [19, 20], [7, 8], [9, 10], [11, 12], [13, 14],
+                 [15, 16], [17, 18]])
+    y = np.array([1, 1, 1, 2, 3, 4, 1, 1, 2, 3, 4, 1, 2, 3, 4])
+    groups = np.array([1, 1, 1, 1, 1, 1, 3, 3, 3, 3, 3, 4, 4, 4, 4])
+    estimator = LogisticRegression(multi_class='multinomial', penalty='l2',
```

Comment:
Hi @jnothman,
specifically, `SGDClassifier` with `incremental` training does not fail without shuffle, but I believe that with shuffle it trains better model (just from looking at the scores):
**Without shuffle:**
```
>>> train_sizes, train_scores, test_scores = learning_curve(estimator, X, y, cv=cv, n_jobs=1, train_sizes=np.linspace(0.3,1.0,3), groups=groups, verbose=1, exploit_incremental_learning=True)
>>> train_scores
array([[ 1.        ,  1.        ],
       [ 0.2       ,  0.2       ],
       [ 0.22222222,  0.16666667]])

```

**With shuffle:**

```
>>> train_sizes, train_scores, test_scores = learning_curve(estimator, X, y, cv=cv, n_jobs=1, train_sizes=np.linspace(0.3,1.0,3), groups=groups, verbose=1, shuffle=True, random_state=2, exploit_incremental_learning=True)
>>> train_scores
array([[ 0.5       ,  0.5       ],
       [ 0.4       ,  0.2       ],
       [ 0.22222222,  0.5       ]])

```